### PR TITLE
scons: use python 3

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by iMichka** [comment]
```
==> brew test scons --verbose
Testing scons
==> /home/linuxbrew/.linuxbrew/Cellar/scons/3.1.2_1/bin/scons
/usr/bin/env: ‘python’: No such file or directory
Error: scons: failed
```

File: Formula/scons.rb [hunk 1]
```diff
@@ -3,26 +3,37 @@ class Scons < Formula
   homepage "https://www.scons.org/"
   url "https://downloads.sourceforge.net/project/scons/scons/3.1.2/scons-3.1.2.tar.gz"
   sha256 "7801f3f62f654528e272df780be10c0e9337e897650b62ddcee9f39fde13f8fb"
+  revision 1 if OS.mac?
```
Comment (dawidd6) [inline Formula/scons.rb]: unless

File: Formula/scons.rb [hunk 2]
```diff
@@ -3,26 +3,37 @@ class Scons < Formula
...
     cellar :any_skip_relocation
     sha256 "eacddf72a0a5803c6053eb8bda0f9b1c0a27f8d7bd30c239c56aa3608802ea15" => :catalina
     sha256 "eacddf72a0a5803c6053eb8bda0f9b1c0a27f8d7bd30c239c56aa3608802ea15" => :mojave
     sha256 "eacddf72a0a5803c6053eb8bda0f9b1c0a27f8d7bd30c239c56aa3608802ea15" => :high_sierra
-    sha256 "3880ad9e8f43d6a5d909042e6a748176382a98707f11b2dc0b51b754539dce4e" => :x86_64_linux
   end
 
-  uses_from_macos "python@2"
+  if OS.mac?
+    uses_from_macos "python@2"
```
Comment (dawidd6) [inline Formula/scons.rb]: Is this necessary?

Comment (iMichka) [reply Formula/scons.rb]: No. I will upstream this patch to homebrew-core right away.

**Pull request merged**